packages/syft/src/syft/core/tensor/autodp/phi_tensor.py

```diff
@@ -2106,6 +2190,25 @@ def sum(
         #     max_vals=max_val,
         # )
 
+    def mean(
+        self,
+        axis: Optional[Union[int, Tuple[int, ...]]] = None,
+        **kwargs: Any,
+    ) -> PhiTensor:
+        child = self.child.mean(axis=axis, **kwargs)
+        data_subjects = np.array(self.data_subjects.mean(axis=axis, **kwargs))
+
+        return PhiTensor(
+            child=child,
+            data_subjects=data_subjects,
+            min_vals=lazyrepeatarray(
+                data=self.min_vals.data, shape=child.shape, data_type="min_val"
```

Comment:
NITPICK: We could do the mean operation on the lazyrepeatarray also.

Reply:
SUGGESTION : Maybe also we could move all the lazyrepeatarray computations to that particular class itself, to maintain modularity like
min_vals = self.min_vals.mean(*args,**kwargs) maybe now or for a future PR.

Reply:
We will definitely have a PR to improve the lazyrepeatarray soon- adding Enums, and using some of the ideas from [here](https://github.com/OpenMined/PySyft/pull/6460) like lazy evaluation (in a safe way).

Reply:
However I do think we should add some comments here in order to make this clear to people in the future

Reply:
Another benefit in this case of not implementing a mean function directly in LazyRepeatArray, but rather creating the object in Phi/GammaTensor is that if we tried to do the former, we would have to call the expensive `to_numpy()` function in order to determine the shape of the output lazyrepeatarray. However, when implemented in the DP Tensor method, we could directly pass on the shape of self.child.mean() as the shape of the lazyrepeatarray, and avoid calling `to_numpy()`.